flake.nix

```diff
@@ -295,6 +295,11 @@
       common-gpu-nvidia-sync = import ./common/gpu/nvidia/prime-sync.nix;
       common-gpu-nvidia-nonprime = import ./common/gpu/nvidia;
       common-gpu-nvidia-disable = import ./common/gpu/nvidia/disable.nix;
+      common-gpu-nvidia-ada-lovelace = import ./common/gpu/nvidia/ada-lovelace;
+      common-gpu-nvidia-ampere = import ./common/gpu/nvidia/ampere;
+      common-gpu-nvidia-turing = import ./common/gpu/nvidia/turing;
+      common-gpu-nvidia-pascal = import ./common/gpu/nvidia/pascal;
+      common-gpu-nvidia-maxwell = import ./common/gpu/nvidia/maxwell;
```

Comment:
Making this a public api, would work against https://github.com/NixOS/nixos-hardware/issues/992 a bit. Progress on that has been slow, but we made good progress on https://github.com/numtide/nixos-facter so I would prefer to not expose stuff in the flake.nix, which we later on have to deprecate again. The alternative would be to make this part of NixOS as well...

With that we could than automatically detect what nvidia generation is installed in a machine.